src/System.Windows.Forms/src/System/Windows/Forms/LinkLabel.cs

```diff
@@ -423,6 +423,22 @@ internal static Rectangle CalcTextRenderBounds(Rectangle textRect, Rectangle cli
 
     protected override AccessibleObject CreateAccessibilityInstance() => new LinkLabelAccessibleObject(this);
 
+    internal override void ReleaseUiaProvider(HWND handle)
+    {
+        if (OsVersion.IsWindows8OrGreater())
+        {
+            base.ReleaseUiaProvider(handle);
+
+            foreach (Link link in _links)
+            {
+                if (link.IsAccessibilityObjectCreated)
+                {
+                    UiaCore.UiaDisconnectProvider(link.AccessibleObject);
+                }
+            }
+        }
```

Comment:
Some code in `base.ReleaseUiaProvider(handle);` still need to be run even if Windows version is less than 8, it's only the `UiaDisconnectProvider` that needs to be guarded. 

```suggestion
        base.ReleaseUiaProvider(handle);

        if (!OsVersion.IsWindows8OrGreater())
        {
            return;
        }

        foreach (Link link in _links)
        {
            if (link.IsAccessibilityObjectCreated)
            {
                UiaCore.UiaDisconnectProvider(link.AccessibleObject);
            }
        }
```